Xamarin.Forms.Platform.UAP/PlatformUWP.cs

```diff
@@ -83,7 +84,32 @@ static void OnPageActionSheet(object sender, ActionSheetArguments options)
 					options.SetResult(null);
 			};
 
-			actionSheet.ShowAt(((Page)sender).GetOrCreateRenderer().ContainerElement);
+			var pageElement = ((Page)sender).GetOrCreateRenderer().ContainerElement;
+			// If pageElement is MainPage it can have pop-up dialogs which overlap this page
+			// In order for ActionSheet to be on top, we need to show it at the dialog box
+			if (pageElement.RenderSize.Height == Window.Current.Bounds.Height
+				&& pageElement.RenderSize.Width == Window.Current.Bounds.Width)
+			{
+				var popups = VisualTreeHelper.GetOpenPopups(Window.Current);
+				foreach (var popup in popups)
+				{
+					if (popup.Child is ContentDialog alert)
+					{
+						actionSheet.ShowAt(alert);
+						return;
+					}
+				}
+			}
```

Comment:
If I comment out lines 90-102, I don't see any different behavior in the test.

Reply:
Strange. Its really works without this code...  Although when the PR was created without it did not work...